core/src/com/github/czyzby/tests/reflected/widgets/CodeTextArea.java

```diff
@@ -0,0 +1,44 @@
+package com.github.czyzby.tests.reflected.widgets;
+
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.kotcrab.vis.ui.widget.VisTextArea;
+
+/**
+ * Custom {@link VisTextArea} used to display LML code in main view. Supports embedding in scrollpane
+ * by calcuating required space needed for current text.
+ * @author Kotcrab
+ */
+public class CodeTextArea extends VisTextArea {
+    private GlyphLayout prefSizeLayout;
+    private String lastText;
+
+    public CodeTextArea(String text, String styleName) {
+        super(text, styleName);
+    }
+
+    @Override
+    protected void initialize() {
+        super.initialize();
+        prefSizeLayout = new GlyphLayout();
+    }
+
+    @Override
+    public float getPrefHeight() {
+        updatePrefSizeLayoutIfNeeded();
+        return prefSizeLayout.height + super.getPrefHeight();
+    }
+
+    @Override
+    public float getPrefWidth() {
+        updatePrefSizeLayoutIfNeeded();
+        return prefSizeLayout.width + super.getPrefHeight();
+    }
+
+    private void updatePrefSizeLayoutIfNeeded() {
+        final String text = getText();
+        if (lastText != text) {
```

Comment:
I'm used to adding comments in situations like that, as it seems like a typo at first. But since it's not an error, it's fine.